.github/actions/setup-cli-deps/action.yml

```diff
@@ -13,7 +13,7 @@ runs:
         git config --global user.name "App Management"
       shell: bash
     - name: Install pnpm
-      uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # pin@v2.2.4
+      uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # pin@v4.0.0
```

Comment:
> We have a policy of pinning github actions to specific commits

Why do we pin to commits instead of versions?

If we were OK with switching to a version reference, `pnpm/action-setup` "supports npm versioning scheme, it could be an exact version (such as `6.24.1`), or a version range (such as `6`, `6.x.x`, `6.24.x`, `^6.24.1`, `*`, etc.), or latest" so we could do `uses: pnpm/action-setup@v4.0.0` to replicate the current change.

However, I wonder why we are so defensive in general. Could we `uses: pnpm/action-setup@v4.0.x` or even better just `uses: pnpm/action-setup@v4`?

Reply:
https://vault.shopify.io/page/GitHub-Actions~13124.md#overview

Reply:
Thanks, Ariel! And I mis-characterized the `version` input for the `pnpm/action-setup` action which is for the version of `pnpm` as the `pnpm/action-setup` action's version; bad job out me!